dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/leader/election/LeaderElectionJob.java

```diff
@@ -0,0 +1,74 @@
+package org.hisp.dhis.leader.election;
+
+/*
+ * Copyright (c) 2004-2018, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import org.hisp.dhis.scheduling.AbstractJob;
+import org.hisp.dhis.scheduling.JobConfiguration;
+import org.hisp.dhis.scheduling.JobType;
+import org.hisp.dhis.system.notification.Notifier;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Job that attempts to elect the current instance as the leader of the cluster.
+ * 
+ * @author Ameen Mohamed
+ */
+public class LeaderElectionJob extends AbstractJob
+{
+    @Autowired
+    private LeaderManager leaderManager;
+
+    @Autowired
+    private Notifier notifier;
+
+    // -------------------------------------------------------------------------
+    // Implementation
+    // -------------------------------------------------------------------------
+
+    @Override
+    public JobType getJobType()
+    {
+        return JobType.LEADER_ELECTION;
+    }
+
+    @Override
+    public void execute( JobConfiguration jobConfiguration )
+    {
+        try
+        {
+            leaderManager.electLeader();
+        }
+        catch ( Exception e )
+        {
+            notifier.notify( jobConfiguration, "Leader election failed:" + e.getMessage() );
+        }
+
+        notifier.notify( jobConfiguration, "Leader election completed" );
```

Comment:
You should propably use "public Notifier notify( JobConfiguration id, NotificationLevel level, String message, boolean completed )" instead so that if someone looks into the job in the notifier they will see that the job completed. In the exception, you can also set NotificationLevel to ERROR

Reply:
Got it!